src/components/dialogs/commons/upload-new-case.tsx

```diff
@@ -5,33 +5,130 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import React from 'react';
-import { FormattedMessage } from 'react-intl';
+import React, { useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
 import Button from '@mui/material/Button';
 import CircularProgress from '@mui/material/CircularProgress';
 import { Grid, Input } from '@mui/material';
-import { useController } from 'react-hook-form';
+import { useController, useFormContext } from 'react-hook-form';
+import {
+    API_CALL,
+    CASE_FILE,
+    CASE_NAME,
+    CASE_UUID,
+} from '../../utils/field-constants';
+import {
+    createCaseWithoutDirectoryElementCreation,
+    deleteCase,
+} from '../../../utils/rest-api';
 
 interface UploadNewCaseProps {
     name: string;
-    caseFileLoading?: boolean;
-    handleCaseFileUpload: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    isNewStudyCreation?: boolean;
+    getCurrentCaseImportParams?: (uuid: string) => void;
+    handleApiCallError?: ErrorCallback;
 }
 
+const MAX_FILE_SIZE_IN_MO = 100;
+const MAX_FILE_SIZE_IN_BYTES = MAX_FILE_SIZE_IN_MO * 1024 * 1024;
+
 const UploadNewCase: React.FunctionComponent<UploadNewCaseProps> = ({
     name,
```

Comment:
Do you really need the name as prop ? It's always `CASE_FILE`.

Reply:
right, It is not necessary